### ISSUE TITLE
Warn if pre-existing file changed

### ISSUE THREAD

**Issue by jakirkham** [conda/conda-build#2762]
Might be nice if `conda-build` could detect a file changed (from a dependency installed in the build environment) when finishing up the build and creating the package. This could aid in detecting conflicts. Would also be nice to have the option to harden this into an error.

If this is already the case and I'm unaware, please feel free to point out the docs and close.

**Comment by msarahan** [comment]
Yes, this would be nice.  It'll go in our backlog unless someone wants to put up a PR.  Shouldn't be too hard - compute hashes for the initial files, then also for the post-build files, and compare for the set that is common to both.  Report mismatches.

**Comment by jakirkham** [comment]
(bumping)
